ContactAlphabeticalSearchUsingMap.cls

```diff
@@ -0,0 +1,44 @@
+public with sharing class ContactAlphabeticalSearchUsingMap {
+    public List<String> alpabetList{get;set;}
+    public String searchAlphabet{get;set;}
+    public list<contact> contactList{set;get;}
+    public list<contact> showContacts{set;get;}
+    public static contact contacts {set; get;}
+    public map<String,List<Contact>> mapAlphabetRecord{get;set;}
+    public String convertedChar;
+    public String query;
+    public ContactAlphabeticalSearchUsingMap()
+    {
+        mapAlphabetRecord = new map<String,List<Contact>>(); 
+        contactList=new List<Contact>([Select name,email,birthdate,mobilePhone from contact order by Name limit 1000]);
+        showContacts = new  list<contact>();
+        alpabetList = new List<String>();
+        List<Integer> charList= new List<Integer>();
+        for (Integer alphabetAsci=65;alphabetAsci<91;alphabetAsci++)
+        {
+            charList.clear();
+            charList.add(alphabetAsci);  
+            String convertedChar = String.fromCharArray(charList); 
+            alpabetList.add(convertedChar);
+            mapAlphabetRecord.put(convertedChar,new List<Contact>());
+        }
+        mapAlphabetRecord.put('All',new List<Contact>());
+        alpabetList.add('All');
+        for(Contact contact : contactList)
+        {
+            mapAlphabetRecord.get('All').add(contact);  
```

Comment:
@Eternus-noopur Why to add contact two times in map? Here you are using unwanted stroage space. Instead when user select `All` that time you code should be like below in display method.
```
for(List<Contact> contList :mapAlphabetRecord.values()){
     showContacts.addAll(contList);
}
```